server/lib/scheduler/Schedule.ts

```diff
@@ -0,0 +1,142 @@
+import { map } from "asyncro";
+import * as _ from "lodash";
+import { UserSchedule } from "./UserSchedule";
+
+class Schedule {
+  // the following object maps root workspace ids to timestamps
+  // I didn't use a Map (and map a workspace object to a timestamp)
+  // because Sequelize doesn't preserve object identity across queries
+  private cacheForWhenTreeLastWorkedOn = {};
```

Comment:
Does this need to have "cache" in its' name? Can it just be something like "lastWorkedOnTime" or similar, with a comment about workspaceId?